### PR TITLE
CI: seperate check-labels and check-commits workflows

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

They were doing completely seperate things.
Also, `check-commits` was unnecessarily re-run for every label change due to shared `on:`.

This'll make it easier to get #19101 in.

### CODE REVIEW

File: .github/workflows/check-labels.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: check-labels
+on:
+  pull_request:
+    types: [opened, reopened, labeled, unlabeled, synchronize]
```
Comment (kaspar030) [inline .github/workflows/check-labels.yml:4]: this might not need "synchronize" anymore

Comment (miri64) [reply .github/workflows/check-labels.yml:4]: Yepp. Not necessary to run on push IMHO.

Comment (kaspar030) [reply .github/workflows/check-labels.yml]: :+1:

(squashed right in)

**Review by miri64 — APPROVED** [review]

**Comment by miri64** [comment]
Does bors still need an update?

**Comment by miri64** [comment]
bors merge

**Comment by kaspar030** [comment]
> Does bors still need an update?

I don't think so, IIUC "check-commits (pr_check)" should still be reported as before.

**Comment by miri64** [comment]
bors merge

**Comment by maribu** [comment]
bors cancel

**Comment by maribu** [comment]
bors merge

**Comment by miri64** [comment]
> bors cancel

Why?

**Comment by maribu** [comment]
> > bors cancel
> 
> Why?

Optimizing CI queues by letting another ACK'ed PR jump onto the merge train. See Matrix room for the discussion. (Also: The cancel and merge cmd can be in the same message, this makes it more reliable that the canceled PR also is added again before the next merge train is started.)

**Comment by miri64** [comment]
> See Matrix room for the discussion. (Also: The cancel and merge cmd can be in the same message, this makes it more reliable that the canceled PR also is added again before the next merge train is started.)

And the reasoning as well, simalar to issue/PR closure. I did not see the Matrix message and chat messages get lost easily in the chatter as well.

**Pull request merged**